portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/openidconnect/OpenIDConnect.testcase

```diff
@@ -0,0 +1,118 @@
+		}
+
+		task ("Enable OpenID Connect login and set Portal's refresh token life time higher than OIDC provider's") {
+			OpenIDConnect.enableOpenIdConnect();
+
+			OpenIDConnect.setTokenRefreshOffset(expirationTime = "120");
+		}
+
+		User.logoutPG();
+
+		task ("Verify that the User is logged out due his token lifespan is over at OIDC provider") {
+			User.loginUserWithOpenIDConnectEnabledPG(
+				userEmailAddress = "user1",
+				userPassword = "test");
+
+			Pause(locator1 = "70000");
+
+			MyAccount.openMyAccountAdmin();
+
+			User.viewLoggedOutPG();
+		}
+	}
+
+	@description = "This is a use case for LPS-124898 TC-1: Verify Instance admin can set the token refresh offset via System settings."
+	@priority = "4"
+	test TokenRefreshOffsetCanBeSetAtSystemLevel {
+		property test.name.skip.portal.instance = "OpenIDConnect#TokenRefreshOffsetCanBeSetAtSystemLevel";
+
+		task ("Enable and set the OpenID connect token refresh offset time") {
+			OpenIDConnect.enableOpenIdConnect();
+
+			OpenIDConnect.setTokenRefreshOffset(expirationTime = "60");
+		}
+	}
+
+	@description = "This is a use case for LPS-124898 TC-5: Verify Token Refresh Offset cannot be saved with a value under 30 second. Blocked by LPS-138193"
+	@priority = "4"
+	test VerifyTokenRefreshOffsetMinimum {
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration~default.config";
+		property portal.upstream = "quarantine";
+		property test.name.skip.portal.instance = "OpenIDConnect#VerifyTokenRefreshOffsetMinimum";
+
+		OpenIDConnect.enableOpenIdConnect();
+
+		OpenIDConnect.setTokenRefreshOffset(expirationTime = "20");
+
+		WaitForConsoleTextPresent(value1 = "Token refresh offset needs to be at least 30 seconds");
+
+		AssertTextEquals(
+			locator1 = "OpenIDConnect#TOKEN_REFRESH_OFFSET",
+			value1 = "30");
+	}
+
+	@description = "This is a use case for LPS-124900 TC-2: User is still logged in after the first access token's lifetime is over and they receive a new access token in the background."
+	@priority = "5"
+	test VerifyUserIsStillLoggedInAfterAccessTokenExpiresAndRenew {
+		property keycloak.enabled = "true";
+		property test.name.skip.portal.instance = "OpenIDConnect#VerifyUserIsStillLoggedInAfterAccessTokenExpiresAndRenew";
+
+		task ("Add and verify that the OSGI configuration has been added") {
+			OpenIDConnect.copyOSGIConnectionConfiguration(OSGiConfigFileName = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration~default.config");
+		}
+
+		task ("Enable OpenID connect") {
+			OpenIDConnect.enableOpenIdConnect();
+		}
+
+		User.logoutPG();
+
+		task ("Verify that the User stay logged in after a new access token is generated") {
+			User.loginUserWithOpenIDConnectEnabledPG(
+				userEmailAddress = "user1",
+				userPassword = "test");
+
+			Pause(locator1 = "65000");
+
+			User.viewLoggedInPG();
```

Comment:
@weller0246 In the original script, we had the order of the following steps

MyAccount.openMyAccountAdmin();
User.viewLoggedInPG();

These 2 step is against each other because in the viewLoggedInPG it checks for files like "email_address" exists,
it will because we did open a page that has email fields present like My Account. This leads us to assertation failure later in the flow when viewLoggedInPG checks for:
[AssertElementNotPresent(locator1 = "TextInput#EMAIL_ADDRESS");](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/macros/User.macro#L2966)

I'm open to a different solution if you have in your mind to resolve this deadlock!

cc:@ferenconodiD